app/models/user.rb

```diff
@@ -0,0 +1,29 @@
+class User < ApplicationRecord
+  validates :full_name, presence: true
+  validates :password, length:{minimum: 8}, format:{with: /(?=.*?[A-Z])(?=(.*[a-z]){1,})(?=(.*[\d]){1,})(?=(.*[\W]){1,})(?!.*\s).{8,}/}
+  validates :full_name, presence: true
+  
+  # Include default devise modules. Others available are:
+  # :timeoutable, and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable, :trackable, :lockable, :confirmable
+
+  attr_writer :login
+
+  def login
+    @login || self.full_name || self.email
+  end
+
+  def self.find_first_by_auth_conditions(warden_conditions)
```

Comment:
Can we rename it like `find_by_auth_conditions`? And are we using this method?

Reply:
Faisal bhai, this is Devise's overridden function that I am using for login through full_name/email so can't rename it.